samples/appendingEvents.go

```diff
@@ -0,0 +1,142 @@
+package samples
+
+import (
+	"context"
+	"log"
+
+	"github.com/EventStore/EventStore-Client-Go/client"
+	"github.com/EventStore/EventStore-Client-Go/messages"
+	"github.com/EventStore/EventStore-Client-Go/options"
+	"github.com/EventStore/EventStore-Client-Go/stream_position"
+	"github.com/EventStore/EventStore-Client-Go/streamrevision"
+)
+
+type TestEvent struct {
+	id            string
+	importantData string
+}
+
+func AppendToStream(db *client.Client) {
+	// region append-to-stream
+	data := TestEvent{
+		id:            "1",
+		importantData: "some value",
+	}
+
+	event, err := messages.NewJsonProposedEvent("some-event", data)
+
+	if err != nil {
+		panic(err)
+	}
+
+	options := options.AppendToStreamOptionsDefault()
+
+	result, err := db.AppendToStream(context.Background(), "some-stream", &options, event)
+	// endregion append-to-stream
+
+	log.Printf("Result: %v", result)
+}
+
+func AppendWithSameId(db *client.Client) {
+	// region append-duplicate-event
+	data := TestEvent{
+		id:            "1",
+		importantData: "some value",
+	}
+
+	event, err := messages.NewJsonProposedEvent("some-event", data)
+
+	if err != nil {
+		panic(err)
+	}
+
+	options := options.AppendToStreamOptionsDefault()
+
+	_, err = db.AppendToStream(context.Background(), "some-stream", &options, event)
+
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = db.AppendToStream(context.Background(), "some-stream", &options, event)
+
+	if err != nil {
+		panic(err)
+	}
+
+	// endregion append-duplicate-event
+}
+
+func AppendWithNoStream(db *client.Client) {
+	// region append-with-no-stream
+	data := TestEvent{
+		id:            "1",
+		importantData: "some value",
+	}
+
+	event, err := messages.NewJsonProposedEvent("some-event", data)
+
+	if err != nil {
+		panic(err)
+	}
+
+	options := options.AppendToStreamOptionsDefault().ExpectedRevision(streamrevision.NoStream())
```

Comment:
This API is a bit strange to me TBH. We need to create some defaults and later override it with some kind of chain builder.

As I mentioned in `options` package, maybe it would be better to just have a struct that can set default values if they are empty. If not, someone can pass this value:

```go
options := options.AppendToStreamOptions{Revision: streamrevision.NoStream()}
```

or even better:

```go
opts := options.AppendToStream{Revision: streamrevision.NoStream()}
```

1. `opts` to not shadow `options` package,
2. `options.AppendToStream` because `options` is already in package name and in Go you are always using types with package name so it's redundant

Reply:
I like that idea but the only issue is some default go values, while not being ESDB default values, are still valid values nonetheless. Detecting those is not possible.

Maybe you are suggesting to use pointers everywhere in those options struct fields?

Reply:
This comment may work in that case as well: https://github.com/EventStore/EventStore-Client-Go/pull/65#discussion_r716940703